lib/ui.ex

```diff
@@ -1,109 +1,157 @@
 defmodule Ui do
 
   @clear_screen "\e[H\e[2J"
-
-    def ask_for_position(board) do
-      IO.puts "Please choose a position:"
-      get_users_position(board)
-    end
-
-    def get_users_position(board) do
-      input = clean_input(IO.gets(""))
-      case valid_position?(input, board) do
-        {:valid, position} -> position
-        {:taken, position} -> move_was_taken(board, position)
-        :not_a_number -> was_not_a_number(board, input)
-      end
-    end
-
-    defp move_was_taken(board, position) do
-      IO.puts "#{position} is already taken!"
-      ask_for_position(board)
-    end
-
-    defp was_not_a_number(board, input) do
-      IO.puts "#{input} is invalid. We need a number!"
-      ask_for_position(board)
-    end
-
-    def play_again? do
-      IO.puts "Play again? (y/n)"
-      get_users_replay_choice
-    end
-
-    def print_board(board) do
-      clear_screen
+  @game_mode [
+    {1, "Human vs Human", :human_vs_human},
+    {2, "Human vs Random", :human_vs_random},
+    {3, "Random vs Human", :random_vs_human},
+  ]
+
+  def ask_for_game_mode do
+    clear_screen
+    IO.puts "Please choose a game mode:\n"
+    print_game_modes(@game_mode, "")
+    get_game_mode
+  end
+
+  defp print_game_modes([], print_options), do: IO.write print_options
+  defp print_game_modes([{number, option, _} | rest], print_options) do
+       updated = print_options <> "#{number} - #{option}\n"
```

Comment:
The indentation is off.
You could have used `#each` if you really wanted. It just struck me as odd that you previously used `#map` which hinted at the fact that `IO.puts` actually returned something useful.